chromium_src/chrome/browser/profiles/profile_avatar_icon_util.cc

```diff
@@ -44,6 +44,13 @@ size_t GetBraveAvatarIconStartIndex();
 #define BRAVE_GET_MODERN_AVATAR_ICON_START_INDEX  \
   return GetBraveAvatarIconStartIndex();
 
+#define BRAVE_GET_ICONS_AND_LABELS_FOR_PROFILE_AVATAR_SELECTOR_NOT_SIGNED_IN \
+  avatars.erase(avatars.begin());                                            \
+  generic_avatar_info = GetDefaultProfileAvatarIconAndLabel_Brave(           \
+      colors.default_avatar_fill_color, colors.default_avatar_stroke_color,  \
+      selected_avatar_idx == GetPlaceholderAvatarIndex());                   \
+  avatars.Insert(avatars.begin(), base::Value(std::move(generic_avatar_info)));
+
```

Comment:
This is not ideal - open to suggestions

Reply:
a possible patchless solution:
```
#define kNotSignedIn kNotSignedIn) {                                          \
    ProfileThemeColors colors = entry->GetProfileThemeColors();               \
    auto generic_avatar_info = GetDefaultProfileAvatarIconAndLabel_Brave(     \
        colors.default_avatar_fill_color, colors.default_avatar_stroke_color, \
        selected_avatar_idx == GetPlaceholderAvatarIndex());                  \
    avatars.Insert(avatars.begin(),                                           \
                   base::Value(std::move(generic_avatar_info)));              \
    return avatars;                                                           \
  } \
  if (false
```

Reply:
>Yes but doesn't that increases the surface area exposed to risky variation from upstream?

I'd say we need to add tests for this part and roll with any of patch solutions. Without tests both options will be fragile in a sense.